internal/services/fluidrelay/fluid_relay_servers_resource_test.go

```diff
@@ -0,0 +1,230 @@
+package fluidrelay_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/fluidrelay/sdk/2022-05-26/fluidrelayservers"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/fluidrelay"
+)
+
+type FluidRelayResource struct{}
+
+func (f FluidRelayResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := fluidrelayservers.ParseFluidRelayServerID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.FluidRelay.ServerClient.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Ptr(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %v", id, err)
+	}
+	if response.WasNotFound(resp.HttpResponse) {
+		return utils.Ptr(false), nil
+	}
+	return utils.Ptr(true), nil
+}
+
+var s = fluidrelay.Server{}
+
+func TestAccFluidRelay_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, s.ResourceType(), "test")
+	f := FluidRelayResource{}
+
+	data.ResourceTest(t, f, []acceptance.TestStep{
+		{
+			Config: f.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(f),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccFluidRelay_userAssigned(t *testing.T) {
+	data := acceptance.BuildTestData(t, s.ResourceType(), "test")
+	f := FluidRelayResource{}
+
+	data.ResourceTest(t, f, []acceptance.TestStep{
+		{
+			Config: f.userAssigned(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(f),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccFluidRelay_systemAssigned(t *testing.T) {
+	data := acceptance.BuildTestData(t, s.ResourceType(), "test")
+	f := FluidRelayResource{}
+
+	data.ResourceTest(t, f, []acceptance.TestStep{
+		{
+			Config: f.systemAssigned(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(f),
+			),
+		},
+		data.ImportStep(),
+	})
+}
```

Comment:
Can we combine these two test cases, together with two other cases: "no identity" and "system and user assigned identity". To test both these 4 cases can be updated from one to antoher?